qt/scientific_interfaces/Indirect/FunctionTemplateBrowser.cpp

```diff
@@ -0,0 +1,117 @@
+#endif
+#endif
+
+#include <QMessageBox>
+#include <QSettings>
+#include <QVBoxLayout>
+#include <limits>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+/**
+ * Constructor
+ * @param parent :: The parent widget.
+ */
+FunctionTemplateBrowser::FunctionTemplateBrowser(QWidget *parent)
+    : QWidget(parent), m_decimals(6) {}
+
+void FunctionTemplateBrowser::createBrowser() {
+  m_stringManager = new QtStringPropertyManager(this);
+  m_doubleManager = new QtDoublePropertyManager(this);
+  m_intManager = new QtIntPropertyManager(this);
+  m_boolManager = new QtBoolPropertyManager(this);
+  m_enumManager = new QtEnumPropertyManager(this);
+  m_groupManager = new QtGroupPropertyManager(this);
+  m_parameterManager = new ParameterPropertyManager(this, true);
+
+  // create editor factories
+  QtSpinBoxFactory *spinBoxFactory = new QtSpinBoxFactory(this);
+  DoubleEditorFactory *doubleEditorFactory = new DoubleEditorFactory(this);
+  QtLineEditFactory *lineEditFactory = new QtLineEditFactory(this);
+  QtCheckBoxFactory *checkBoxFactory = new QtCheckBoxFactory(this);
+  QtEnumEditorFactory *comboBoxFactory = new QtEnumEditorFactory(this);
+  auto *doubleDialogFactory = new DoubleDialogEditorFactory(this, true);
+
+  m_browser = new QtTreePropertyBrowser(nullptr, QStringList(), false);
+  // assign factories to property managers
+  m_browser->setFactoryForManager(m_stringManager, lineEditFactory);
+  m_browser->setFactoryForManager(m_doubleManager, doubleEditorFactory);
+  m_browser->setFactoryForManager(m_intManager, spinBoxFactory);
+  m_browser->setFactoryForManager(m_boolManager, checkBoxFactory);
+  m_browser->setFactoryForManager(m_enumManager, comboBoxFactory);
+  m_browser->setFactoryForManager(m_parameterManager, doubleDialogFactory);
+
+  connect(m_intManager, SIGNAL(propertyChanged(QtProperty *)), this,
+          SLOT(intChanged(QtProperty *)));
+  connect(m_boolManager, SIGNAL(propertyChanged(QtProperty *)), this,
+          SLOT(boolChanged(QtProperty *)));
+  connect(m_enumManager, SIGNAL(propertyChanged(QtProperty *)), this,
+          SLOT(enumChanged(QtProperty *)));
+  connect(m_parameterManager, SIGNAL(propertyChanged(QtProperty *)), this,
+          SLOT(parameterChanged(QtProperty *)));
+
+  connect(doubleDialogFactory, SIGNAL(buttonClicked(QtProperty *)), this,
+          SLOT(parameterButtonClicked(QtProperty *)));
+  connect(doubleDialogFactory, SIGNAL(closeEditor()), m_browser,
+          SLOT(closeEditor()));
+
+  m_browser->setContextMenuPolicy(Qt::CustomContextMenu);
+  connect(m_browser, SIGNAL(customContextMenuRequested(const QPoint &)), this,
+          SLOT(popupMenu(const QPoint &)));
+
+  connect(m_browser, SIGNAL(optionChanged(QtProperty *, const QString &, bool)),
+          this, SLOT(globalChanged(QtProperty *, const QString &, bool)));
+}
+
+void FunctionTemplateBrowser::init() {
+  createBrowser();
+  createProperties();
+  QVBoxLayout *layout = new QVBoxLayout(this);
```

Comment:
As discussed offline QtWidgets own there child objects and take charge of deleting them, it is better to use raw pointers here to avoid double deleting objects.